Store.js

```diff
@@ -140,14 +151,168 @@ define(["dcl/dcl", "decor/Invalidating"], function (dcl, Invalidating) {
 						collection.on("delete", this._itemRemoved.bind(this));
 						collection.on("refresh", this._refreshHandler.bind(this));
 					}
-					return this.processCollection(collection);
+				} else {
+					for (var i = 0; i < this.store.length; i++) {
+						if (this._isItemObservable(this.store[i])) {
+							this._hasItemObservable = true;
+							this._observeResults =  [];
+							// affect the callback to the observe function if the item is observable
+							this._observeResults[i + 1] = Observable.observe(this.store[i],
+								this.observeCallback.bind(this));
+						}
+					}
+					collection = processQueryResult.call(this, this.store.filter(this._isQueried, this));
+					if (this._isArrayObservable(this.store)) {
+						if (!this._observeResults) {
+							this._observeResults = [];
+						}
+						// affect the callback to the observe function if the array is observable
+						this._observeResults[0] = ObservableArray.observe(this.store, this.observeCallback.bind(this));
+					}
 				}
+				return this.processCollection(collection);
 			} else {
 				this.initItems([]);
 			}
 		},
 
 		/**
+		 * Synchronously deliver change records to all listeners registered via `observe()`.
+		 */
+		deliver: dcl.superCall(function (sup) {
+			return function () {
+				sup.call();
+				if (this._observeResults) {
+					this._observeResults[0].deliver();
+					if (this._hasItemObservable) {
+						this.observeCallback && Observable.deliverChangeRecords(this.observeCallback);
```

Comment:
What you need to know about `Function.prototype.bind()` in this context is that it returns a new (bound) function instead of changing the given function. In short:

```javascript
var fn = function () {};
alert(fn.bind({}) === fn); // false
```